server.go

```diff
@@ -114,6 +116,15 @@ func NewServer() *Server {
 		shardsBySeriesID: make(map[uint32][]*Shard),
 		Logger:           log.New(os.Stderr, "[server] ", log.LstdFlags),
 	}
+	// Server will always return with authentication enabled.
```

Comment:
Perhaps add "This ensures that disabling authentication must be an explicit decision"?